### PR TITLE
TS autest extension can now auto select both ssl and nonssl port

### PR DESCRIPTION
**Pull request by macisasandwich**
Along with test modifications to take advantage of this

### CODE REVIEW

**Comment by macisasandwich** [comment]
Reopened PR because I accidentally clicked close PR on the old one

File: tests/gold_tests/tls_hooks/tls_hooks16.test.py [hunk 1]
```diff
@@ -70,9 +69,7 @@
 
 ts.Streams.stderr = "gold/ts-client-hello-1.gold"
 
-snistring = "Client Hello callback 0"
-ts.Streams.All = Testers.ContainsExpression(
-    "\A(?:(?!{0}).)*{0}(?!.*{0}).*\Z".format(snistring), "Client Hello message appears only once", reflags=re.S | re.M)
+#snistring = "Client Hello callback 0"
```
Comment (randall) [inline tests/gold_tests/tls_hooks/tls_hooks16.test.py]: Was this intended to be commited? If so, can it just be removed?

File: tests/tools/plugins/ssl_hook_test.cc [hunk 2]
```diff
@@ -295,7 +295,7 @@ setup_callbacks(TSHttpTxn txn, int preaccept_count, int client_hello_count, int
   TSDebug(PN, "Setup callbacks pa=%d client_hello=%d client_hello_imm=%d sni=%d cert=%d cert_imm=%d pa_delay=%d", preaccept_count,
           client_hello_count, client_hello_count_immediate, sni_count, cert_count, cert_count_immediate, preaccept_count_delay);
   for (i = 0; i < preaccept_count; i++) {
-    cb = TSContCreate(&CB_Pre_Accept, TSMutexCreate());
```
Comment (randall) [inline tests/tools/plugins/ssl_hook_test.cc]: Is this supposed to be part of this change?

File: tests/gold_tests/tls/tls_client_versions.test.py [hunk 3]
```diff
@@ -41,9 +41,7 @@
 ts.addSSLfile("ssl/server.pem")
 ts.addSSLfile("ssl/server.key")
 
-ts.Variables.ssl_port = 4443
-
-# Need no remap rules.  Everything should be proccessed by sni
+# Need no remap rules.  Everything should be proccessed by ssl_server_name
```
Comment (randall) [inline tests/gold_tests/tls/tls_client_versions.test.py]: this should be sni

File: tests/gold_tests/tls/tls_forward_nonhttp.test.py [hunk 4]
```diff
@@ -22,15 +22,13 @@
...
+ts = Test.MakeATSProcess("ts", select_ports=True, enable_tls=True)
 
 # add ssl materials like key, certificates for the server
 ts.addSSLfile("ssl/server.pem")
 ts.addSSLfile("ssl/server.key")
 
-ts.Variables.ssl_port = 4443
-
-# Need no remap rules.  Everything should be proccessed by sni
+# Need no remap rules.  Everything should be proccessed by ssl_server_name
```
Comment (randall) [inline tests/gold_tests/tls/tls_forward_nonhttp.test.py]: this should be sni

File: tests/gold_tests/tls/tls_tunnel.test.py [hunk 5]
```diff
@@ -45,9 +45,7 @@
 ts.addSSLfile("ssl/signer.pem")
 ts.addSSLfile("ssl/signer.key")
 
-ts.Variables.ssl_port = 4443
-
-# Need no remap rules.  Everything should be proccessed by sni
+# Need no remap rules.  Everything should be proccessed by ssl_server_name
```
Comment (randall) [inline tests/gold_tests/tls/tls_tunnel.test.py]: this should be sni

File: tests/gold_tests/tls/tls_tunnel_forward.test.py [hunk 6]
```diff
@@ -47,9 +47,7 @@
 ts.addSSLfile("ssl/signer.pem")
 ts.addSSLfile("ssl/signer.key")
 
-ts.Variables.ssl_port = 4443
-
-# Need no remap rules.  Everything should be proccessed by sni
+# Need no remap rules.  Everything should be proccessed by ssl_server_name
```
Comment (randall) [inline tests/gold_tests/tls/tls_tunnel_forward.test.py]: This (and all mentions of ssl_server_name in this changeset) should be sni

**Comment by masaori335** [comment]
Please update tests under [/tests/gold_tests/h2/](https://github.com/apache/trafficserver/tree/06a8bc4c85944572c17040b718c4d3d4517d9fe7/tests/gold_tests/h2) too. They're using ssl ports for HTTP/2 over TLS. Separated PR is fine if you want keep changes small.

**Comment by macisasandwich** [comment]
@masaori335 I will do another PR for those.

File: tests/gold_tests/tls_hooks/tls_hooks16.test.py (hunk 1 above)
Comment (macisasandwich) [reply tests/gold_tests/tls_hooks/tls_hooks16.test.py]: done

File: tests/tools/plugins/ssl_hook_test.cc (hunk 2 above)
Comment (macisasandwich) [reply tests/tools/plugins/ssl_hook_test.cc]: Yeah this is an intended change.

File: tests/gold_tests/tls/tls_client_versions.test.py (hunk 3 above)
Comment (macisasandwich) [reply tests/gold_tests/tls/tls_client_versions.test.py]: changed all the instances of ssl_server_name to sni

**Comment by macisasandwich** [comment]
@randall can you mark this and its sibling #5660 for 8.x backport?

**Comment by masaori335** [comment]
This and #5660 is added to 8.x releases projects for backport candidate.

**Pull request merged**